File: backend/pkg/models/files.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/datatypes"
)

type FileIndex struct {
	ID            uuid.UUID      `json:"id" gorm:"type:uuid;primaryKey;default:uuid_generate_v4()"`
	S3Key         string         `json:"s3_key" gorm:"not null;uniqueIndex"`
	Filename      string         `json:"filename" gorm:"not null"`
	DirectoryPath string         `json:"directory_path"`
	FileType      string         `json:"file_type"`
	FileSize      int64          `json:"file_size"`
	ContentType   string         `json:"content_type"`
	MetaData      datatypes.JSON `json:"meta_data" gorm:"type:jsonb;default:'{}'"`
	CreatedAt     time.Time      `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt     time.Time      `json:"updated_at" gorm:"autoUpdateTime"`
}

func (FileIndex) TableName() string {
	return "file_index"
}

File: backend/src/database/models/files.rs
use chrono::{DateTime, Utc};
use diesel::{Identifiable, Insertable, Queryable, Selectable};
use serde_json::Value;
use uuid::Uuid;

use crate::schema::file_index;

#[derive(Debug, Clone, Queryable, Selectable, Identifiable)]
#[diesel(table_name = file_index)]
#[diesel(check_for_backend(diesel::pg::Pg))]
pub struct FileIndexRow {
    pub id: Uuid,
    pub s3_key: String,
    pub filename: String,
    pub directory_path: Option<String>,
    pub file_type: Option<String>,
    pub file_size: Option<i64>,
    pub content_type: Option<String>,
    pub meta_data: Option<Value>,
    pub created_at: Option<DateTime<Utc>>,
    pub updated_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, Insertable)]
#[diesel(table_name = file_index)]
pub struct NewFileIndexRow {
    pub id: Uuid,
    pub s3_key: String,
    pub filename: String,
    pub directory_path: Option<String>,
    pub file_type: Option<String>,
    pub file_size: Option<i64>,
    pub content_type: Option<String>,
    pub meta_data: Option<Value>,
}
